File: Crassus.Native.iOS/TextFieldDelegate.m
//
//  TextFieldDelegate.m
//  TheObiG LLC
//
//  Created by SolomonRain on 2/15/16.
//  Copyright © 2016 Crassus. All rights reserved.
//

#import <Foundation/Foundation.h>
#import <UIKit/UIKit.h>
#import "TextFieldDelegate.h"
#import "GlobalDefinitions.h"
#import "Crassus_Native_iOS-Swift.h"

void* CrassusCreateTextField(Frame frame)
{
    return (void*)CFBridgingRetain([CrassusTextField create:frame]);
}

void CrassusSetTextFieldText(void* textField, const char * text)
{
    ((__bridge UITextField*)textField).text = [NSString stringWithUTF8String:text];
}

void CrassusSetPlaceholder(void* textField, const char * placeholder)
{
    ((__bridge UITextField*)textField).placeholder = [NSString stringWithUTF8String:placeholder];
}

void CrassusSetTextFieldBorderColor(void* textField, Rgb color)
{
    ((__bridge UITextField*)textField).layer.borderColor = [UIColor colorWithRed:color.red green:color.green blue:color.blue alpha:1.0].CGColor;
}

void CrassusSetTextFieldBorderWidth(void* textField, float width)
{
    ((__bridge UITextField*)textField).layer.borderWidth = width;
}

void CrassusSetTextFieldCornerRadius(void* textField, float radius)
{
    ((__bridge UITextField*)textField).layer.cornerRadius = radius;
}

void CrassusSetTextFieldLeftView(void* textField, void* view)
{
    ((__bridge UITextField*)textField).leftView = CFBridgingRelease(view);
}

void CrassusSetTextFieldViewMode(void* textField, TextFieldViewMode mode)
{
    if (mode == TextFieldViewModeAlways) ((__bridge UITextField*)textField).leftViewMode = UITextFieldViewModeAlways;
    if (mode == TextFieldViewModeNever) ((__bridge UITextField*)textField).leftViewMode = UITextFieldViewModeNever;
    if (mode == TextFieldViewModeUnlessEditing) ((__bridge UITextField*)textField).leftViewMode = UITextFieldViewModeUnlessEditing;
    if (mode == TextFieldViewModeWhileEditing) ((__bridge UITextField*)textField).leftViewMode = UITextFieldViewModeWhileEditing;
}

void CrassusSetSecureEntry(void* textField, Bools secure)
{
    ((__bridge UITextField*)textField).secureTextEntry = secure == BoolsYES;
}

File: Crassus.Native.iOS/TextFieldDelegate.c
//
//  TextFieldDelegate.c
//  TheObiG LLC
//
//  Created by SolomonRain on 2/15/16.
//  Copyright © 2016 Crassus. All rights reserved.
//

#include "TextFieldDelegate.h"
#include "GlobalDefinitions.h"

void* CreateTextField(Frame frame, ViewCreator function)
{
    return (*function)(frame);
}

void SetTextFieldText(void* textField, const char * text, ObjectTextAction action)
{
    (*action)(textField, text);
}

void SetPlaceholder(void* textField, const char * placeholder, ObjectTextAction action)
{
    (*action)(textField, placeholder);
}

void SetTextFieldBorderColor(void* textField, Rgb color, ViewShortColorAction action)
{
    (*action)(textField, color);
}

void SetTextFieldBorderWidth(void* textField, float width, ObjectDecimalAction action)
{
    (*action)(textField, width);
}

void SetTextFieldCornerRadius(void* textField, float radius, ObjectDecimalAction action)
{
    (*action)(textField, radius);
}

void SetTextFieldLeftView(void* textField, void* view, DoubleObjectAction action)
{
    (*action)(textField, view);
}

void SetTextFieldViewMode(void* textField, TextFieldViewMode mode, ViewModeAction action)
{
    (*action)(textField, mode);
}

void SetSecureEntry(void* textField, Bools secure, ObjectToggler action)
{
    (*action)(textField, secure);
}